crates/hir_def/src/item_tree/lower.rs

```diff
@@ -190,11 +190,15 @@ impl Ctx {
                             },
                             ast::Item(item) => {
                                 // FIXME: This triggers for macro calls in expression position
-                                let mod_items = self.lower_mod_item(&item, true);
-                                let current_block = block_stack.last();
-                                if let (Some(mod_items), Some(block)) = (mod_items, current_block) {
-                                    if !mod_items.0.is_empty() {
-                                        self.data().inner_items.entry(*block).or_default().extend(mod_items.0.iter().copied());
+                                if let Some(SyntaxKind::MACRO_TYPE) = node.parent().map(|p| p.kind()) {
```

Comment:
Does this fix the FIXME above?

Reply:
I dont think so, why do you need to touch this code at all?

Reply:
Because macros at certain type positions (eg. `let a: U32!() = ...`) hit this path then further down the line panics here: https://github.com/rust-analyzer/rust-analyzer/blob/c0de9ed71e74833013bf73601c376f6ade03d82e/crates/hir_def/src/item_tree.rs#L113

Reply:
Please do that in `file_item_tree_query`, there are other special cases just like this there

Reply:
Oh. Funny thing, I did scroll though that code several times but did not even nice that lol. I'll add another case for that when I get home :)

Reply:
Done.